enterprise/frontend/src/embedding-sdk-bundle/components/public/MetabotQuestion/MetabotQuestion.tsx

```diff
@@ -1,100 +1,102 @@
-import { useId } from "react";
-import { t } from "ttag";
+import { useId, useMemo } from "react";
+import { match } from "ts-pattern";
 
-import {
-  SdkLoader,
-  withPublicComponentWrapper,
-} from "embedding-sdk-bundle/components/private/PublicComponentWrapper";
+import { FlexibleSizeComponent } from "embedding-sdk-bundle/components/private/FlexibleSizeComponent";
+import { withPublicComponentWrapper } from "embedding-sdk-bundle/components/private/PublicComponentWrapper";
 import { SdkAdHocQuestion } from "embedding-sdk-bundle/components/private/SdkAdHocQuestion";
 import { SdkQuestionDefaultView } from "embedding-sdk-bundle/components/private/SdkQuestionDefaultView";
 import { EnsureSingleInstance } from "embedding-sdk-shared/components/EnsureSingleInstance/EnsureSingleInstance";
+import useIsSmallScreen from "metabase/common/hooks/use-is-small-screen";
 import { useLocale } from "metabase/common/hooks/use-locale";
-import { Flex, Paper, Stack, Text } from "metabase/ui";
-import { Messages } from "metabase-enterprise/metabot/components/MetabotChat/MetabotChatMessage";
-import { MetabotResetLongChatButton } from "metabase-enterprise/metabot/components/MetabotChat/MetabotResetLongChatButton";
-import {
-  useMetabotAgent,
-  useMetabotChatHandlers,
-} from "metabase-enterprise/metabot/hooks";
+import { Stack } from "metabase/ui";
 import { useMetabotReactions } from "metabase-enterprise/metabot/hooks/use-metabot-reactions";
 
-import { MetabotChatEmbedding } from "./MetabotChatEmbedding";
+import { MetabotChatHistory } from "./MetabotChatHistory";
+import { MetabotChatInput } from "./MetabotChatInput";
+import { MetabotChatSuggestions } from "./MetabotChatSuggestions";
+import S from "./MetabotQuestion.module.css";
 import { metabotQuestionSchema } from "./MetabotQuestion.schema";
+import { MetabotQuestionEmptyState } from "./MetabotQuestionEmptyState";
 import { QuestionDetails } from "./QuestionDetails";
 import { QuestionTitle } from "./QuestionTitle";
-
-const MetabotQuestionInner = () => {
+import { SidebarHeader } from "./SidebarHeader";
+import type { MetabotQuestionProps } from "./types";
+
+const MetabotQuestionInner = ({
+  height,
+  width,
+  className,
+  style,
+  layout = "auto",
+}: MetabotQuestionProps) => {
   const { isLocaleLoading } = useLocale();
   const { navigateToPath } = useMetabotReactions();
-
-  if (isLocaleLoading) {
-    return <SdkLoader />;
+  const isSmallScreen = useIsSmallScreen();
```

Comment:
Maybe better to use content-queries?
I assume it may be embedded inside a container and then better to rely on container dimensions

Reply:
Agree! I was thinking the same but couldn't find a `useContainerQuery` hook (something like [this](https://github.com/joshuaaron/use-container-queries/blob/main/src/index.ts)); I tried to do it via CSS but it needs a lot of styles duplication for the automatic layout and the stacked/sidebar layout, so I'd prefer using JS here. Let me see if I can do something with the element width 🙏🏻

Reply:
Fixed in https://github.com/metabase/metabase/pull/64142/commits/7fe7538220e0efefed7318c6789b30877cec8b01, thanks!